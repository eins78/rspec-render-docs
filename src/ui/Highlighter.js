import React from 'react'

const Highlighter = ({ mark, children }) =>
  mark ? <mark>{children}</mark> : <span>{children}</span>

export default Highlighter
