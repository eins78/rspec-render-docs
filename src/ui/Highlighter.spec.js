/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Highlighter from './Highlighter'

it('renders', () => {
  const component = renderer.create(
    <div>
      <Highlighter mark>marked text</Highlighter>
      <hr />
      <Highlighter mark={false}>not marked text</Highlighter>
      <hr />
      <Highlighter>not marked text</Highlighter>
    </div>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
