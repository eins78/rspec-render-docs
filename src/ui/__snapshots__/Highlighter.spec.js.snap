exports[`test renders 1`] = `
<div>
  <mark>
    marked text
  </mark>
  <hr />
  <span>
    not marked text
  </span>
  <hr />
  <span>
    not marked text
  </span>
</div>
`;
